environment.go

```diff
@@ -738,21 +726,46 @@ func (e *envState) Close(ctx context.Context) error {
 	catcher := grip.NewBasicCatcher()
 	wg := &sync.WaitGroup{}
 	for n, closer := range e.closers {
-		if closer == nil {
+		if !closer.background {
```

Comment:
I don't see the need to have some be background, since the only difference is that some are ran serially and some in parallel. Why do some closers need separate handling?